code/modules/vehicles/mecha/combat/justice.dm

```diff
@@ -0,0 +1,455 @@
+	for_line_turf.Beam(src, icon_state = "mech_charge", time = 8)
+	forceMove(finish_turf)
+
+/obj/vehicle/sealed/mecha/justice/melee_attack_effect(mob/living/victim, heavy)
+	if(!heavy)
+		victim.Knockdown(40)
+		return
+	var/obj/item/bodypart/cut_bodypart = victim.get_bodypart(pick(BODY_ZONE_R_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_ARM, BODY_ZONE_L_LEG))
+	if(prob(DISMEMBER_CUALITY_HIGTH))
+		cut_bodypart.dismember(BRUTE)
+
+
+/obj/vehicle/sealed/mecha/justice/mob_exit(mob/M, silent, randomstep, forced)
+	. = ..()
+	if(!stealth_action?.on)
+		return
+	animate(src, alpha = 255, time = 0.5 SECONDS)
+	playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+
+/obj/vehicle/sealed/mecha/justice/Bump(atom/obstacle)
+	. = ..()
+	if(!isliving(obstacle))
+		return
+	if(!stealth_action?.on)
+		return
+	animate(src, alpha = 255, time = 0.5 SECONDS)
+	playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+
+/obj/vehicle/sealed/mecha/justice/Bumped(atom/movable/bumped_atom)
+	. = ..()
+	if(!isliving(bumped_atom))
+		return
+	if(!stealth_action?.on)
+		return
+	animate(src, alpha = 255, time = 0.5 SECONDS)
+	playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+
+/obj/vehicle/sealed/mecha/justice/take_damage(damage_amount, damage_type, damage_flag, sound_effect, attack_dir, armour_penetration)
+	if(stealth_action?.on)
+		animate(src, alpha = 255, time = 0.5 SECONDS)
+		playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+	if(LAZYLEN(occupants))
+		if(prob(60))
+			new /obj/effect/temp_visual/mech_sparks(get_turf(src))
+			playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+			return
+	return ..()
+
+/datum/action/vehicle/sealed/mecha/invisibility
+	name = "Invisibility"
+	button_icon_state = "mech_stealth_off"
+	/// Is invisibility activated.
+	var/on = FALSE
+	/// Recharge check.
+	var/charge = TRUE
+	/// Block movement if we start aoe attack from invisibility.
+	var/start_attack = FALSE
+	/// Aoe pre attack sound.
+	var/stealth_pre_attack_sound = 'sound/mecha/mech_stealth_pre_attack.ogg'
+	/// Aoe attack sound.
+	var/stealth_attack_sound = 'sound/mecha/mech_stealth_attack.ogg'
+
+/datum/action/vehicle/sealed/mecha/invisibility/Trigger(trigger_flags)
+	. = ..()
+	if(chassis.weapons_safety)
+		owner.balloon_alert(owner, "safty is on!")
+		return
+	if(!charge)
+		owner.balloon_alert(owner, "on recharge!")
+		return
+	new /obj/effect/temp_visual/mech_sparks(get_turf(chassis))
+	on = !on
+	playsound(chassis, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+	if(on)
+		for(var/mob/living/occupant in chassis.occupants)
+			var/datum/action/vehicle/sealed/mecha/charge_attack/charge_action = LAZYACCESSASSOC(chassis.occupant_actions, occupant, /datum/action/vehicle/sealed/mecha/charge_attack)
+			if(charge_action?.on)
+				charge_action.on = FALSE
+		animate(chassis, alpha = 0, time = 0.5 SECONDS)
+		button_icon_state = "mech_stealth_on"
+		addtimer(CALLBACK(src, PROC_REF(end_stealth)), 20 SECONDS)
```

Comment:
It's a shame that mechs can't have status effects because that'd be the ideal interface.